### PR TITLE
Cpe screensaver

### PR DESCRIPTION
**Pull request by cgerke**
Pull? or you guys got some other kinda magic on its way?

### CODE REVIEW

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 1]
```diff
@@ -50,10 +67,94 @@
...
-                      node['cpe_screensaver']['askForPasswordDelay']
+                    'askForPassword' => node['cpe_screensaver']['askForPassword'],
+                    'askForPasswordDelay' => node['cpe_screensaver']['askForPasswordDelay']
+                  }
+                }
+              ]
+            }
+          }
+        },
+        {
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]: should these be added every time? What if i dont want to manage the screeen save type ?

**Comment by cgerke** [comment]
Better? or were you subtly trying to tell me to go away ;)

File: chef/cookbooks/cpe_screensaver/README.md [hunk 2]
```diff
@@ -24,3 +28,18 @@ For example, you could tweak the above values
...
+    # KenBurns, Classic OR a custom path '/Users/YOURUSERNAME/Pictures'
+    node.default['cpe_screensaver']['styleKey']
+
+    # Acceptable (defaults to 1-National Geographic) values: 1-National Geographic, 2-Aerial, 3-Cosmos, 4-Nature Patterns
+    node.default['cpe_screensaver']['SelectedFolderPath'] = 1-National Geographic
+
+    # Acceptable (defaults to 0) values: 0  or 1
+    node.default['cpe_screensaver']['ShufflesPhotos'] = 0
+
+Although you can manage the screensaver "module" via a profile, the UI doesn't get locked down. This could be misleading to the end user, they may think they can change it via the UI and will find out that their settings keep getting mashed by the profile. Consider a lockdown profile "DisabledPreferencePanes" to work in conjuntion with this cookbook.
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/README.md]: 80 cols
do you have rubocop and foodcritic installed ? Please run them against your changes

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 3]
```diff
@@ -11,9 +11,24 @@
 # LICENSE file in the root directory of this source tree. An additional grant
 # of patent rights can be found in the PATENTS file in the same directory.
 #
+module_name = ''
+
+prefix = node['cpe_profiles']['prefix']
+organization = node['organization'] ? node['organization'] : 'Facebook'
+screensaver_profile = {
+  'PayloadIdentifier' => "#{prefix}.screensaver",
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]: not runtime safe. This need to be in the ruby_block to ensure this gets assigned during runtime

File: chef/cookbooks/cpe_screensaver/attributes/default.rb [hunk 4]
```diff
@@ -15,3 +15,7 @@
 default['cpe_screensaver']['idleTime'] = 600
 default['cpe_screensaver']['askForPassword'] = 1
 default['cpe_screensaver']['askForPasswordDelay'] = 0
+default['cpe_screensaver']['MESSAGE'] = ''
+default['cpe_screensaver']['SelectedFolderPath'] = ''
+default['cpe_screensaver']['styleKey'] = ''
+default['cpe_screensaver']['ShufflesPhotos'] = 0
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/attributes/default.rb]: default['cpe_screensaver']['MESSAGE'] = nil
default['cpe_screensaver']['SelectedFolderPath'] = nil
default['cpe_screensaver']['styleKey'] = nil
default['cpe_screensaver']['ShufflesPhotos'] = nil

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 5]
```diff
@@ -25,42 +40,148 @@
...
-    prefix = node['cpe_profiles']['prefix']
-    organization = node['organization'] ? node['organization'] : 'Facebook'
-    node.default['cpe_profiles']["#{prefix}.screensaver"] = {
-      'PayloadIdentifier' => "#{prefix}.screensaver",
-      'PayloadRemovalDisallowed' => true,
-      'PayloadScope' => 'System',
-      'PayloadType' => 'Configuration',
-      'PayloadUUID' => 'CEA1E58D-9D0F-453A-AA52-830986A8366C',
-      'PayloadOrganization' => organization,
+    unless node['cpe_screensaver']['MESSAGE'].empty? && node['cpe_screensaver']['SelectedFolderPath'].empty?
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]: unless node['cpe_screensaver']['MESSAGE'].nil? && node['cpe_screensaver']['SelectedFolderPath'].nil?

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 6]
```diff
@@ -25,42 +40,148 @@
...
+                'askForPassword' => node['cpe_screensaver']['askForPassword'],
+                'askForPasswordDelay' => node['cpe_screensaver']['askForPasswordDelay']
+              }
+            }
+          ]
+        }
+      }
+    })
+
+    unless node['cpe_screensaver']['MESSAGE'].to_s == ''
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]: unless node['cpe_screensaver']['MESSAGE']

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 7]
```diff
@@ -25,42 +40,148 @@
...
+      message = node['cpe_screensaver']['MESSAGE'] ? node['cpe_screensaver']['MESSAGE'] : organization
+      module_name = 'Computer Name'
+      screensaver_profile['PayloadContent'].push({
+        'PayloadType' => 'com.apple.ManagedClient.preferences',
+        'PayloadVersion' => 1,
+        'PayloadIdentifier' => "#{prefix}.screensaver.Basic",
+        'PayloadUUID' => '8dd9a983-f59b-47c3-a408-ac55d287cbd4',
+        'PayloadEnabled' => true,
+        'PayloadDisplayName' => 'MESSAGE',
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]: should this be:
'PayloadDisplayName' => node['cpe_screensaver']['MESSAGE'],

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 8]
```diff
@@ -25,42 +40,148 @@
...
+                  'MESSAGE' => message
+                }
+              }
+            ]
+          }
+        }
+      })
+    end
+
+    unless module_name == '' && node['cpe_screensaver']['SelectedFolderPath'].to_s == ''
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]: unless module_name.nil? && node['cpe_screensaver']['SelectedFolderPath']

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 9]
```diff
@@ -25,42 +40,148 @@
...
+                'idleTime' => node['cpe_screensaver']['idleTime'],
+                'askForPassword' => node['cpe_screensaver']['askForPassword'],
+                'askForPasswordDelay' => node['cpe_screensaver']['askForPasswordDelay']
+              }
+            }
+          ]
+        }
+      }
+    })
+
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]: module_name = nil

File: chef/cookbooks/cpe_screensaver/recipes/default.rb [hunk 10]
```diff
@@ -25,42 +40,148 @@
...
-    node.default['cpe_profiles']["#{prefix}.screensaver"] = {
-      'PayloadIdentifier' => "#{prefix}.screensaver",
-      'PayloadRemovalDisallowed' => true,
-      'PayloadScope' => 'System',
-      'PayloadType' => 'Configuration',
-      'PayloadUUID' => 'CEA1E58D-9D0F-453A-AA52-830986A8366C',
-      'PayloadOrganization' => organization,
+    unless node['cpe_screensaver']['MESSAGE'].empty? && node['cpe_screensaver']['SelectedFolderPath'].empty?
+      Chef::Log.warn(
+        'Screensaver module management has conflicting keys!'
```
Comment (mikedodge04) [inline chef/cookbooks/cpe_screensaver/recipes/default.rb]:
```
fail 'Screensaver module management has conflicting keys! MESSAGE && SelectedFolderPath '
```

**Comment by cgerke** [comment]
Thanks for all the tips (I just started playing with Ruby). Hopefully a bit tighter now. Should I squash the commits to make the history nicer?

**Comment by natewalck** [comment]
Github has been pretty good about squashing for us, so I am cool with it as is. I'll look at this again on Monday and poke Mike.

**Comment by cgerke** [comment]
Cool.if there is a better way to do any of this, happy to take suggestions. Im new to Ruby.

**Comment by cgerke** [comment]
Wondering if a part of this or all of it should move to a resource too now? Only issue being I assume you want guyes [idleTime / askForPassword / askForPasswordDelay] to always run.

**Comment by erikng** [comment]
Everything should be in a resource, yes.

On Aug 22, 2016, at 10:55 PM, cgerke <notifications@github.com<mailto:notifications@github.com>> wrote:


Wondering if a part of this or all of it should move to a resource too now? Only issue being I assume you want guyes [idleTime / askForPassword / askForPasswordDelay] to always run.

-

**Comment by cgerke** [comment]
ok will have a crack at it. only issue i see is those items that probably shouldnt allow an override by the user if they were to set them to nil so i'd probably have to split this into a resource per domain?

**Comment by natewalck** [comment]
Here is the plan. Mike is going to land a custom resource version of this, then I'll have you rebase these changes on the new version and we can code review from there.

**Comment by natewalck** [comment]
Ok, this needs to be manually rebased now that  #46 has landed. If you run into any issues/confusion, let me know.

**Comment by cgerke** [comment]
Think we are there.

Added another option for node.default['cpe_screensaver']['SelectedFolderPath']

You can specify a string value 'FILENAME.saver'

 which assumes the 3rd party is in the OS path '/Library/Screen Savers/FILENAME.saver'

**Comment by natewalck** [comment]
Github is saying there are still conflicts ;(

**Comment by cgerke** [comment]
Odd my branch shows no conflicts? What do you see?

**Comment by cgerke** [comment]
Wait, I see it. Few mins.

**Comment by cgerke** [comment]
Done. I think.

**Comment by cgerke** [comment]
Should have some time tonight to clean it up.

**Comment by cgerke** [comment]
I'm redoing this. Its mangled.

**Pull request closed without merging**